engine/runtime/src/main/scala/org/enso/compiler/context/ChangesetBuilder.scala

```diff
@@ -42,8 +43,43 @@ final class ChangesetBuilder[A: TextEditor: IndexedSource](
     * @return the computed changeset
     */
   @throws[CompilerError]
-  def build(edits: Seq[TextEdit]): Changeset[A] =
-    Changeset(source, ir, compute(edits))
+  def build(edits: Seq[TextEdit]): Changeset[A] = {
+
+    def findSimpleChange(): (IR.Literal, TextEdit) = {
+      val edit = if (edits.size != 1) {
```

Comment:
it feels like you should be pattern matching on the seq rather than checking  the size and doing `.head`/`.tail`

Reply:
@JaroslavTulach you can do this kind of stuff in Scala
```scala
scala> Seq(1) match {
     | case Seq(x)         => Seq(x, x, x)
     | case Seq(x, y)      => Seq(x, y, y)
     | case x +: y +: rest => Seq(x, y) ++ rest
     | }
val res12: Seq[Int] = List(1, 1, 1)
```

Reply:
Done in [a84ae04](https://github.com/enso-org/enso/pull/3508/commits/a84ae0425070de185501ea02fc54cbc9e0ab7188)